arch/arm64/boot/dts/qcom/sm8475-samsung-q4q.dts

```diff
@@ -0,0 +1,738 @@
+		vdd-s2-supply = <&vph_pwr>;
+		vdd-s3-supply = <&vph_pwr>;
+
+		vdd-l1-l2-supply = <&pmr735a_s2>;
+		vdd-l3-supply = <&pmr735a_s1>;
+		vdd-l4-supply = <&pm8350c_s1>;
+		vdd-l5-l6-supply = <&pm8350c_s1>;
+		vdd-l7-bob-supply = <&vreg_bob>;
+		*/
+
+		pmr735a_s2: smps2 {
+			regulator-name = "pmr735a_s2";
+			regulator-min-microvolt = <500000>;
+			regulator-max-microvolt = <1040000>;
+		};
+
+		pmr735a_s3: smps3 {
+			regulator-name = "pmr735a_s3";
+			regulator-min-microvolt = <300000>;
+			regulator-max-microvolt = <2352000>;
+		};
+
+		pmr735a_l1: ldo1 {
+			regulator-name = "pmr735a_l1";
+			regulator-min-microvolt = <800000>;
+			regulator-max-microvolt = <880000>;
+		};
+
+		pmr735a_l2: ldo2 {
+			regulator-name = "pmr735a_l2";
+			regulator-min-microvolt = <480000>;
+			regulator-max-microvolt = <1200000>;
+		};
+
+		pmr735a_l3: ldo3 {
+			regulator-name = "pmr735a_l3";
+			regulator-min-microvolt = <1200000>;
+			regulator-max-microvolt = <1200000>;
+		};
+
+		pmr735a_l4: ldo4 {
+			regulator-name = "pmr735a_l4";
+			regulator-min-microvolt = <1776000>;
+			regulator-max-microvolt = <1776000>;
+		};
+
+		pmr735a_l5: ldo5 {
+			regulator-name = "pmr735a_l5";
+			regulator-min-microvolt = <880000>;
+			regulator-max-microvolt = <920000>;
+		};
+
+		pmr735a_l6: ldo6 {
+			regulator-name = "pmr735a_l6";
+			regulator-min-microvolt = <1200000>;
+			regulator-max-microvolt = <1200000>;
+		};
+
+		pmr735a_l7: ldo7 {
+			regulator-name = "pmr735a_l7";
+			regulator-min-microvolt = <2800000>;
+			regulator-max-microvolt = <2800000>;
+		};
+	};
+};
+
+&gpi_dma0 {
+	status = "okay";
+};
+
+&gpi_dma1 {
+	status = "okay";
+};
+
+&gpi_dma2 {
+	status = "okay";
+};
+
+&i2c5 {
+	clock-frequency = <100000>;
+	status = "okay";
+
+	/* nq@64 rtc6226 */
+	/* fsa4480@42 qcom,fsa4480-i2c */
+	/* redriver@1c onnn,redriver */
+	/* aw20036_led@3a awinic,aw20036_led */
+	/* eusb2_repeater@4f nxp,eusb2-repeater */
+
+	nxp_eusb2_repeater: eusb2_repeater@4f {
+		compatible = "nxp,eusb2-repeater";
+		reg = <0x4f>;
+		vdd18-supply = <&pm8350_s10>;
+		vdd3-supply = <&pm8350_l2>;
+		reset-gpio = <&pm8350c_gpios 7 GPIO_ACTIVE_HIGH>;
+		pinctrl-names = "default";
+		pinctrl-0 = <&eusb2_reset_ctrl_default>;
+		#phy-cells = <0>;
+		qcom,param-override-seq =
+				/* Rx squelch detection threshold to 110mV; default is 125mV */
+				<0x40 0x06
+				/*
+				 * Tx Deemphasis to 2dB, Tx Deemphasis bit duration to 0.8UI;
+				 * default is 0 for both
+				 */
+				0x22 0x07
+				/* Output Voltage Swing to 550mV; default is 450mV */
+				0x64 0x08>;
+	};
+};
+
+&i2c9 {
+	clock-frequency = <1000000>;
+	status = "okay";
+
+	/* nq@28(ts) qcom,sn-nci */
+};
+
+&i2c13 {
+	clock-frequency = <1000000>;
+	status = "okay";
+
+	/* tfa98xx@34 */
+	/* tfa98xx@35 */
+};
+
+&i2c18 {
+	clock-frequency = <1000000>;
+	status = "okay";
+
+	/* haptic_hv@5a awinic,aw8692x */
+};
+
+&pm8350c_gpios {
+	eusb2_reset_ctrl_default: eusb2_reset_ctrl_default {
+		pins = "gpio7";
+		function = "normal";
```

Comment:
function = PMIC_GPIO_FUNC_NORMAL; (This hasnt been fixed in my dts yet, but I'll do it later)

Reply:
#include <dt-bindings/pinctrl/qcom,pmic-gpio.h> Probably needed